backend/tests/worker.rs

```diff
@@ -961,7 +961,7 @@ fn spawn_test_worker(
     tokio::sync::broadcast::Sender<()>,
     tokio::task::JoinHandle<()>,
 ) {
-    for x in [windmill_worker::LOCK_CACHE_DIR] {
+    for x in [windmill_worker::LOCK_CACHE_DIR, windmill_worker::GO_BIN_CACHE_DIR] {
```

Comment:
Seems weird that we need to treat Go differently here, and TBH I don't quite unerstand why we don't need to to it for the other runtimes (we do it in the main.rs, just not for the test worker). But if we don't do that, this line will fail: https://github.com/windmill-labs/windmill/pull/2652/files#diff-9f2b0f927e0a13c103a4e641c255cc58e4f745cc4cafafb28a7296e8c7f9a1baR213

Reply:
Actually, thinking more about it, it might be b/c Go is our only compiled runtime. Anyway, I haven't looked too much into the other runtimes implementation of caching, but curious to get your thoughts after your review

Reply:
I believe in main.rs we create those cache dir so it's just that we were too lazy to mimic proper initialization